common/containers/Tabs/SendTransaction/components/WalletInfo.tsx

```diff
@@ -0,0 +1,143 @@
+    if (this.props.wallet !== nextProps.wallet) {
+      this.setWalletAsyncState(nextProps.wallet);
+    }
+  }
+
+  public render() {
+    const { address, privateKey, isPrivateKeyVisible, isKeystoreModalOpen } = this.state;
+
+    return (
+      <div className="WalletInfo">
+        <div className="Tab-content-pane">
+          <div className="row form-group">
+            <div className="col-xs-11">
+              <label>{translate('x_Address')}</label>
+              <input className="form-control" disabled={true} value={address} />
+            </div>
+            <div className="col-xs-1" style={{ padding: 0 }}>
+              <Identicon address={address} />
+            </div>
+          </div>
+
+          {privateKey && (
+            <div className="row form-group">
+              <div className="col-xs-12">
+                <label>{translate('x_PrivKey')}</label>
+                <div className="input-group">
+                  <input
+                    className="form-control"
+                    disabled={true}
+                    type={isPrivateKeyVisible ? 'text' : 'password'}
+                    value={privateKey}
+                  />
+                  <span
+                    onClick={this.togglePrivateKey}
+                    aria-label={translateRaw('GEN_Aria_2')}
+                    role="button"
+                    className="input-group-addon eye"
+                  />
+                </div>
+              </div>
+            </div>
+          )}
+        </div>
+
+        <div className="Tab-content-pane">
+          <div className="row">
+            <div className="col-xs-6">
+              <label>Public Address</label>
+              <div className="WalletInfo-qr well well-lg">
+                <QRCode data={address} />
+              </div>
+            </div>
+            {privateKey && (
+              <div className="col-xs-6">
+                <label>Private Key</label>
+                <div className="WalletInfo-qr well well-lg" onClick={this.togglePrivateKey}>
+                  <QRCode data={isPrivateKeyVisible ? privateKey : '0'} />
+                  {!isPrivateKeyVisible && (
+                    <div className="WalletInfo-qr-cover">
+                      <i className="WalletInfo-qr-cover-icon fa fa-eye" />
+                    </div>
+                  )}
+                </div>
+              </div>
+            )}
+
+            {privateKey && (
+              <div className="col-xs-6">
+                <label>Utilities</label>
+
+                <button className="btn btn-info btn-block" onClick={print(address, privateKey)}>
+                  {translate('x_Print')}
+                </button>
+
+                <button className="btn btn-info btn-block" onClick={this.toggleKeystoreModal}>
+                  Generate Keystore File
+                </button>
+              </div>
+            )}
+
+            {isKeystoreModalOpen && (
+              <GenerateKeystoreModal
```

Comment:
Does it make more sense to just pass `isOpen` as a prop to the modal? I saw that you just keep the original Modal `isOpen` prop to `true`, instead it could  be passed through

Reply:
This has the nice side-effect of clearing the modal's state every time you close it since it un-renders it, so the fields return to blank. But I'll add it as a prop to GenerateKeystoreModal and pass it through, for any future use cases.